specification/eventhub/resource-manager/Microsoft.EventHub/preview/2022-01-01-preview/ApplicationGroups.json

```diff
@@ -0,0 +1,349 @@
+          },
+          {
+            "$ref": "../../../common/v1/definitions.json#/parameters/NamespaceNameParameter"
+          },
+          {
+            "$ref": "../../../common/v2/definitions.json#/parameters/ApplicationGroupNameParameter"
+          },
+          {
+            "$ref": "../../../common/v1/definitions.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "../../../common/v1/definitions.json#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Namespace ApplicationGroup returned successfully.",
+            "schema": {
+              "$ref": "#/definitions/ApplicationGroup"
+            }
+          },
+          "default": {
+            "description": "Eventhub error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../common/v2/definitions.json#/definitions/ErrorResponse"
+            }
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "ApplicationGroupListResult": {
+      "type": "object",
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ApplicationGroup"
+          },
+          "description": "Result of the List Application Groups operation."
+        },
+        "nextLink": {
+          "type": "string",
+          "description": "Link to the next set of results. Not empty if Value contains an incomplete list of Authorization Rules"
+        }
+      },
+      "description": "The response from the List Application Groups operation."
+    },
+    "ApplicationGroup": {
+      "type": "object",
+      "properties": {
+        "properties": {
+          "type": "object",
+          "x-ms-client-flatten": true,
+          "properties": {
+            "updatedAt": {
+              "readOnly": true,
+              "format": "date-time",
+              "type": "string",
+              "description": "Exact time the Application Group was updated"
+            },
+            "createdAt": {
+              "readOnly": true,
+              "format": "date-time",
+              "type": "string",
+              "description": "Exact time the Application Group was created."
+            },
+            "isEnabled": {
+              "type": "boolean",
+              "description": "Determines if Application Group is allowed to create connection with namespace or not. Once the isEnabled is set to false, all the existing connections of application group gets dropped and no new connections will be allowed"
+            },
+            "clientAppGroupIdentifier": {
+              "type": "string",
+              "description": "The Unique identifier for application group.Supports SAS(NamespaceSASKeyName=KeyName or EntitySASKeyName=KeyName) or AAD(AADAppID=Guid)"
+            },
+            "policies": {
```

Comment:
>policies [](http://example.com/codeflow?start=13&length=8)

instead of an array why not just model it as  policies.rateLimitThreshold.IncomingBytes
policies.rateLimitThreshold.OutgoingBytes
policies.rateLimitThreshold.IncomingMessages
?

Isnt it cleaner this way and you can also save on a lot of validations you might have to implement if you leave it as an array?

Reply:
I think you are coming from the perspective that there can only be 1 policy based on the metric Id which is applicable? Till now we were going under the logic that even even several policies are set for a metric Id only 1 of them(the lowest) one will be applied.

Reply:
@yogeshgargmsft @kasun04

Reply:
@rkmanda This is also a smart approach. But it does not look extendable if we have to add more properties related to "throttlingPolicy" for the same metricId e.g if we want to add "throttlingApplicableTill" timestamp field and has the flexibility that it can be applied at individual metricId, then we loose out on that extendibility.

With arrays, yes we need to have certain checks & balances which will be present in service code and will keep getting added as the feature matures.

Reply:
Ok please be aware that either ways when you add new policies, they will need to be added in a new API version.  API version 1 will have to populate the policies specific to that version and API version 2 will have to add a different set on a GET verb.

The other issue with arrays is that when the client wants to modify only a single element, they still have to ensure theres logic in place to manipulate the whole array. So in general we are discouraging teams from using arrays. If you are willing to deal with the validation/ versioning complexities it is fine but pl do so with the understanding of all the implications.